optuna/trial/_trial.py

```diff
@@ -485,6 +485,21 @@ def objective(trial):
             )
             raise TypeError(message) from None
 
+        try:
+            if not isinstance(step, int):
+                warnings.warn(
+                    "The reported step is going to cast to int because `step`s type is {}.".format(
+                        type(step).__name__
+                    )
+                )
```

Comment:
I personally prefer `type(step)` rather than `type(step).__name__` as it is more informative when we get something like `np.int32`.
```suggestion
                warnings.warn(
                    f"Got step with type {type(step)} in report, but will be cast to int."
                )
```

Reply:
Thanks. Shall we change `value`'s error too? Maybe as a followup.